src/meta/src/rpc/ddl_controller.rs

```diff
@@ -1801,12 +1832,32 @@ impl DdlController {
             .get_job_fragments_by_id(&id.into())
             .await?;
         let old_internal_table_ids = old_fragments.internal_table_ids();
-        let old_internal_tables = self
-            .metadata_manager
-            .get_table_catalog_by_ids(old_internal_table_ids)
-            .await?;
 
-        fragment_graph.fit_internal_table_ids(old_internal_tables)?;
+        // handle drop table's associated source
+        let mut release_ctx = None;
+        if drop_table_associated_source_id.is_some() {
+            // drop table's associated source means the fragment containing the table has just one internal table (associated source's state table)
+            debug_assert!(old_internal_table_ids.len() == 1);
+
+            release_ctx = Some(ReleaseContext {
+                database_id: stream_job.database_id() as i32,
+                // we do not remove the original table catalog as it's still needed for the streaming job
+                // just need to remove the ref to the state table
+                removed_streaming_job_ids: vec![id as i32],
+                removed_source_ids: vec![drop_table_associated_source_id.unwrap() as i32],
+                removed_state_table_ids: old_internal_table_ids
+                    .iter()
+                    .map(|id| *id as i32)
+                    .collect(),
+                ..Default::default()
```

Comment:
It's still a little strange to use `ReleaseContext` here, as I said before that `removed_streaming_job_ids` is not to remove the id but to be updated. It is better to introduce a new context struct, which is clearer and meaningful.

Reply:
Introduced `DropTableConnectorContext` to replace it.